### PR TITLE
Introduced EditableLimitableEntry widget

### PR DESCRIPTION
**Pull request by abalakh**
```python
pytest -v tests/foreman/ui_airgun/test_activationkey.py -k test_negative_usage_limit
============================= test session starts ==============================
platform darwin -- Python 3.6.4, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /Users/andrii/workspace/py3a/bin/python3.6
cachedir: .cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: wait-for-1.0.9, xdist-1.22.0, services-1.2.1, mock-1.6.3, forked-0.2
collected 6 items
2018-03-23 15:30:47 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui_airgun/test_activationkey.py::test_negative_usage_limit PASSED [100%]

============================== 5 tests deselected ==============================
=================== 1 passed, 5 deselected in 312.34 seconds ===================
```

### CODE REVIEW

**Comment by abalakh** [comment]
```python
pytest -v tests/foreman/ui_airgun/test_activationkey.py -k "test_negative_usage_limit or test_positive_create"
============================= test session starts ==============================
platform darwin -- Python 3.6.4, pytest-3.3.2, py-1.5.2, pluggy-0.6.0 -- /Users/andrii/workspace/py3a/bin/python3.6
cachedir: .cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: /Users/andrii/workspace/robottelo, inifile:
plugins: wait-for-1.0.9, xdist-1.22.0, services-1.2.1, mock-1.6.3, forked-0.2
collected 6 items
2018-03-26 14:09:28 - conftest - DEBUG - BZ deselect is disabled in settings


tests/foreman/ui_airgun/test_activationkey.py::test_positive_create PASSED [ 33%]
tests/foreman/ui_airgun/test_activationkey.py::test_positive_create_with_lce_and_cv PASSED [ 66%]
tests/foreman/ui_airgun/test_activationkey.py::test_negative_usage_limit PASSED Made common LimitInput widget
[100%]

============================== 3 tests deselected ==============================
=================== 3 passed, 3 deselected in 624.87 seconds ===================
```

File: airgun/widgets.py [hunk 1]
```diff
@@ -549,3 +598,8 @@ class EditableEntrySelect(EditableEntry):
     but by select list
     """
     edit_field = Select(locator=".//select")
+
+
+class EditableLimitEntry(EditableEntry):
+    """:class:`EditableEntry` which contains :class:`LimitInput` inside."""
```
Comment (oshtaier) [inline airgun/widgets.py]: fix docstring

File: airgun/widgets.py [hunk 2]
```diff
@@ -490,6 +491,54 @@ def fill(self, value):
...
+            return False
+        if str(value).lower() == 'unlimited':
+            self.unlimited.fill(True)
+        else:
+            self.unlimited.fill(False)
+            self.limit.fill(value)
+        return True
+
+    def read(self):
+        """Return either 'Unlimited' if corresponding checkbox is checked off
```
Comment (oshtaier) [inline airgun/widgets.py]: `Return either 'Unlimited' if corresponding checkbox is selected`

File: airgun/widgets.py [hunk 3]
```diff
@@ -490,6 +491,54 @@ def fill(self, value):
...
+            ".//input[@type='checkbox'][contains(@ng-required, 'unlimited') "
+            "or contains(@ng-model, 'unlimited')]"))
+    limit = TextInput(
+        locator=(
+            ".//input[@type='number'][contains(@ng-required, 'unlimited')]"))
+
+    def fill(self, value):
+        """Handle 'Unlimited' checkbox before trying to fill text input.
+
+        :param value: either 'Unlimited' (case insensitive) to check off
```
Comment (oshtaier) [inline airgun/widgets.py]: `to select`

File: airgun/widgets.py [hunk 4]
```diff
@@ -490,6 +491,54 @@ def fill(self, value):
...
+    Example html representation::
+
+        <input type="checkbox" name="limit"
+         ng-model="activationKey.unlimited_hosts"...>
+
+        <input id="max_hosts" name="max_hosts"
+         ng-model="activationKey.max_hosts"
+         ng-required="!activationKey.unlimited_hosts" type="number" min="1"
+         max="2147483648"...>
+
```
Comment (oshtaier) [inline airgun/widgets.py]: missed mandatory section
```
     Locator example::

```

Comment (abalakh) [reply airgun/widgets.py]: As there's no locator. And can't be any - 2 actual widgets inside are located in different parts of DOM, so if we were accepting locators, we'd probably had to go with smth like `LimitInput(locator1='foo', locator2='bar')`. That was my initial concern i've told you, that's why i'm not really happy about such approach.

Comment (oshtaier) [reply airgun/widgets.py]: cool, add that information to that section

Comment (abalakh) [reply airgun/widgets.py]: done

**Pull request merged**